### PR TITLE
Adds a new `Triggers` property to ModuleBase

### PR DESCRIPTION
**Pull request by kog**
* ModuleBase will default this to be the `Name` property
    * This means anyone not defining a custom trigger will use the module name (e.g. about, help)
    * Or, for things like Factoids, we can use custom triggers (e.g. idiot, warn, ban)
    * These are a prefix: ModuleBase will do `startsWith(triggerPrefix + trigger)` where the prefix is from your config (default: `~`)
    * This is done in an `Any()` across all potential triggers

### CODE REVIEW

File: src/SpikeCore/SpikeCore/Modules/ModuleBase.cs [hunk 1]
```diff
@@ -16,16 +15,24 @@ public abstract class ModuleBase : IMessageHandler<IrcChannelMessageMessage>, IM
         public abstract string Description { get; }
         public abstract string Instructions { get; }
 
+        public virtual List<string> Triggers => new List<string> {Name};
```
Comment (AndyCJ) [inline src/SpikeCore/SpikeCore/Modules/ModuleBase.cs]: Would it be a good idea to make this IEnumerable<string> so that implementers have more flexibility in what provides the values?

Comment (kog) [reply src/SpikeCore/SpikeCore/Modules/ModuleBase.cs]: I initially made it a List because I thought I was going to do something with ordering. I wound up just doing an Any and a regex on the other end, so in retrospect I guess we don't need a list.

TL;DR Summary: yeah, sure, this makes sense.
